### PR TITLE
[FLINK-14862][runtime] Fuse initializeState and open

### PR DESCRIPTION
**Pull request by igalshilman**
## What is the purpose of the change

This PR fuses the two distinct initialization and open passes done at `StreamTask#init`  into a single pass.
Previously `AbstractStreamOperator#initalizeState` and `AbstractStreamOperator#open` are called from two distinct loops (tail to head). Fusing these loops into a single loop allows operators chained together to emit down stream already on initalizeState.

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / no ****/ don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java [hunk 1]
```diff
@@ -968,13 +967,14 @@ private void checkpointState(
 		checkpointingOperation.executeCheckpointing();
 	}
 
-	private void initializeState() throws Exception {
+	private void initializeStateAndOpen() throws Exception {
```
Comment (aljoscha) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java]: The `openAllOperators()` method is now unused and should be removed. We should also move the comment that explains the order of initialization/opening from `openAllOperators()` to this method.

**Comment by igalshilman** [comment]
@kl0u As far as I can see the operator or the task lifecycle hasn't really changed, the same steps still exists logically. What kind of change would you propose to the documentation ?

**Comment by kl0u** [comment]
@igalshilman Actually you are right. No changes required.

**Comment by aljoscha** [comment]
There were checkstyle errors. I squashed the commits and fixed them to see what travis does with it.

**Pull request merged**